lib/node/node.js

```diff
@@ -449,7 +457,11 @@ module.exports.create = function createNode(type, expectedParams, options) {
         // `beforeCreate` allows to change/modify the node before it's actually created/returned
         // if it throws an error it will fail to create the node.
         if (options.beforeCreate) {
-            options.beforeCreate.bind(this)();
+            this.beforeCreate = options.beforeCreate.bind(this);
```

Comment:
Because they serve different purposes. The `beforeCreate` validates the input params in an async manner, or let you setup the analysis node from anything that is not async. Look at it almost as a constructor. Some examples:
 - https://github.com/CartoDB/camshaft/blob/5fdd0b57a05a4c71531c300f757eec73783c1c8b/lib/node/nodes/source.js#L10-L19 
- https://github.com/CartoDB/camshaft/blob/5fdd0b57a05a4c71531c300f757eec73783c1c8b/lib/node/nodes/georeference-city.js#L16-L24

The purpose of the new async method is to allow you to run actual async code, as proposed in the PR description:
> Allow executing an async function before the node is created. This allows preparing the node using metadata from the database.